torchgeo/trainers/detection.py

```diff
@@ -21,6 +21,9 @@
 
 from ..datasets.utils import unbind_samples
 
+plt.switch_backend("agg")
```

Comment:
This shouldn't be here, we don't importing this file to inadvertently change matplotlib backends.